### PR TITLE
feat(helm): add --format flag to 'helm list'

### PR DESCRIPTION
**Pull request by rbwsam**
When 'helm list --format (text|json|yaml) is run, this will print the list
of releases in the specified format.

WIP #1534

Once this is approved I will create more PRs adding this functionality to other commands.

### CODE REVIEW

**Comment by jascott1** [comment]
Hi @rbwsam, 
Thanks for the PR. We have discussed this elsewhere as `-o | --output` to be consistent with `kubectl` and also we have a prior PR that we need to reconcile with your work here. https://github.com/kubernetes/helm/pull/2836

**Comment by rbwsam** [comment]
@jascott1 `-o` is already being used for `-o, --offset` in `helm list`. What flag do you suggest we use for this situation?

**Comment by jascott1** [comment]
@rbwsam Sorry I didn't realize that. I guess we can roll with your original suggestion unless someone else has another idea. The `-o` inconsistency will bug me forever :/

**Comment by alexey-medvedchikov** [comment]
:+1: 
Waiting for this PR to be merged: I've implemented kubectl `go-template`-like behavior for it.

**Comment by mrgleeco** [comment]
is this PR DOA? This is a killer feature..

**Comment by rbwsam** [comment]
If there are any changes required so this can get merged I am happy to oblige.

**Comment by bacongobbler** [comment]
Thasnks @rbwsam. We were all at KubeCon and then ran straight into the holidays. I'm finally coming out of holiday mode. I'll take a look at this now. Thanks for your patience!

File: cmd/helm/list.go [hunk 1]
```diff
@@ -207,7 +223,26 @@ func (l *listCmd) statusCodes() []release.Status_Code {
...
 
-func formatList(rels []*release.Release) string {
+func formatList(rels []*release.Release, format string) (string, error) {
+	var res string
+	var err error
+
+	switch format {
+	case "text":
+		res = formatText(rels)
+	case "short":
```
Comment (bacongobbler) [inline cmd/helm/list.go]: can you add this to the list of supported output formats in the docstring?

File: cmd/helm/list.go [hunk 2]
```diff
@@ -150,19 +165,20 @@ func (l *listCmd) run() error {
...
-		return nil
+		format = "short"
+	}
+
+	formatted, err := formatList(res.Releases, format)
+	if err != nil {
+		return err
 	}
-	fmt.Fprintln(l.out, formatList(rels))
+	fmt.Fprint(l.out, formatted)
```
Comment (bacongobbler) [inline cmd/helm/list.go]: have you tested this change with the text and short output formats? Looks fine, but just want to make sure that this has been tested.

File: cmd/helm/list_test.go [hunk 3]
```diff
@@ -48,6 +48,22 @@ func TestListCmd(t *testing.T) {
...
+			expected: "\"Name\": \"atlas\",\n",
+		},
+		{
+			name: "list format YAML",
+			args: []string{"--format", "yaml"},
+			resp: []*release.Release{
+				releaseMock(&releaseOptions{name: "atlas"}),
+			},
+			expected: "- Chart: foo-0.1.0-beta.1\n  Name: atlas\n  Namespace: default\n  Revision: 1\n  Status: DEPLOYED\n",
+		},
```
Comment (bacongobbler) [inline cmd/helm/list_test.go]: can we add a "short" list format test as well?

**Review by bacongobbler — CHANGES_REQUESTED** [review]

**Comment by bacongobbler** [comment]
ping, looks like this PR needs a rebase as well as the above comments. We're cutting v2.8.0 next week so if you'd like to get this PR in, now's a good time to fix it up :)

**Comment by mattfarina** [comment]
@rbwsam I'd like to get this into Helm 2.9. I'm happy to support you in cleaning it up or in taking over the work myself.

**Comment by michelleN** [comment]
ping @rbwsam - We'd love to see this go through. Any luck on addressing those comments above?

**Comment by alexey-medvedchikov** [comment]
Have some work made upon this PR so will make another PR with lunapark and stuff if original poster won't appear in nerest time :)

**Comment by seaneagan** [comment]
Since there wasn't much activity here of late and we have been looking forward to this functionality I went ahead and took a shot at this in #3923.

@rbwsam

**Comment by au-phiware** [comment]
Can this be closed now, since #3923?

**Comment by bacongobbler** [comment]
yes, thank you @au-phiware for pointing that out. :)

**Pull request closed without merging**